contracts/nexus-gateway/src/contract/execute.rs

```diff
@@ -1,17 +1,54 @@
 use connection_router::state::Message;
-use cosmwasm_std::Response;
-use error_stack::Result;
+use cosmwasm_std::{Addr, Response};
+use error_stack::ResultExt;
 
 use crate::error::ContractError;
 use crate::nexus;
+use crate::state::Store;
 
-pub fn route_to_nexus(msgs: Vec<Message>) -> Result<Response<nexus::Message>, ContractError> {
-    let msgs = msgs
-        .into_iter()
-        .map(TryInto::try_into)
-        .collect::<Result<Vec<nexus::Message>, _>>()?;
+use super::Contract;
 
-    Ok(Response::new().add_messages(msgs))
+type Result<T> = error_stack::Result<T, ContractError>;
+
+impl<S> Contract<S>
+where
+    S: Store,
+{
+    pub fn route_messages(
+        mut self,
+        sender: Addr,
+        msgs: Vec<Message>,
+    ) -> Result<Response<nexus::Message>> {
+        match sender {
+            sender if sender == self.config.nexus => todo!(),
+            sender if sender == self.config.router => self
+                .route_to_nexus(msgs)
+                .change_context(ContractError::RouteToNexus),
+            _ => Err(ContractError::Unauthorized)?,
```

Comment:
it's sad that this has to be a conditional match instead of a direct equality comparison, but I still prefer this match pattern over an `if.. else if` chain